### ISSUE TITLE
训练脚本运行报错main.py: error: unrecognized arguments: --local-rank=1

### ISSUE THREAD

**Issue by 6Roy** [OFA-Sys/Chinese-CLIP#94]
您好，我希望在单机6卡训练，然而一直报错，不知道哪出错了
if true; then
    nohup python -u -m torch.distributed.launch --nproc_per_node=${GPUS_PER_NODE}   \
        --master_addr=${MASTER_ADDR} --master_port=${MASTER_PORT} /home/yuxiang/Chinese-CLIP/cn_clip/training/main.py \
        --train-data=${train_data} \
        --val-data=${val_data} \
        --resume=${resume} \
        ${reset_data_offset} \
        ${reset_optimizer} \
        --logs=${output_base_dir} \
        --name=${name} \
        --save-step-frequency=${save_step_frequency} \
        --save-epoch-frequency=${save_epoch_frequency} \
        --log-interval=${log_interval} \
        ${report_training_batch_acc} \
        --context-length=${context_length} \
        --warmup=${warmup} \
        --batch-size=${batch_size} \
        --valid-batch-size=${valid_batch_size} \
        --valid-step-interval=${valid_step_interval} \
        --valid-epoch-interval=${valid_epoch_interval} \
        --accum-freq=${accum_freq} \
        --lr=${lr} \
        --wd=${wd} \
        --max-epochs=${max_epochs} \
        --vision-model=${vision_model} \

**Comment by 6Roy** [comment]
Pytorch是2.0.0

**Comment by DtYXs** [comment]
您好，或许您可以尝试一下1.10.1版本，应该是正常运行的

**Comment by DtYXs** [comment]
您好，或许您可以尝试以下1.10版本，应该是正常运行的
如果您想在2.0.0版本运行，可以进行以下代码改动后再尝试运行：
1.将.sh脚本中`python3 -m torch.distributed.launch`改为`torchrun`
https://github.com/OFA-Sys/Chinese-CLIP/blob/0925e08e53b239559da6477b8fbbde62130ea15c/run_scripts/muge_finetune_vit-b-16_rbt-base.sh#L60
2.将`cn_clip/training/main.py`中第51行`args.local_device_rank = max(args.local_rank, 0)`改为`args.local_device_rank = os.int(os.environ['LOCAL_RANK'])`
https://github.com/OFA-Sys/Chinese-CLIP/blob/0925e08e53b239559da6477b8fbbde62130ea15c/cn_clip/training/main.py#L51

**Comment by wktxdy** [comment]
/cn_clip/training/params.py
添加：
parser.add_argument("--local_rank", type=int, default=1)

**Comment by hito-Chen** [comment]
> 您好，或许您可以尝试一下1.10版本，应该是正常运行的 如果您想在2.0.0版本运行，可以进行以下代码改动后再尝试运行： 1.将.sh脚本中`python3 -m torch.distributed.launch`改为`torchrun`
> 
> [Chinese-CLIP/run_scripts/muge_finetune_vit-b-16_rbt-base.sh](https://github.com/OFA-Sys/Chinese-CLIP/blob/0925e08e53b239559da6477b8fbbde62130ea15c/run_scripts/muge_finetune_vit-b-16_rbt-base.sh#L60)
> 
> Line 60 in [0925e08](/OFA-Sys/Chinese-CLIP/commit/0925e08e53b239559da6477b8fbbde62130ea15c)
> 
>  python3 -m torch.distributed.launch --nproc_per_node=${GPUS_PER_NODE} --nnodes=${WORKER_CNT} --node_rank=${RANK} \ 
> 
> 2.将`cn_clip/training/main.py`中第51行`args.local_device_rank = max(args.local_rank, 0)`改为`args.local_device_rank = int(os.environ['LOCAL_RANK'])`
> [Chinese-CLIP/cn_clip/training/main.py](https://github.com/OFA-Sys/Chinese-CLIP/blob/0925e08e53b239559da6477b8fbbde62130ea15c/cn_clip/training/main.py#L51)
> 
> Line 51 in [0925e08](/OFA-Sys/Chinese-CLIP/commit/0925e08e53b239559da6477b8fbbde62130ea15c)
> 
>  args.local_device_rank = max(args.local_rank, 0)

老师你好，我已经按照您说的进行修改，但是还是一样的报错，
运行脚本：
![Image](https://github.com/user-attachments/assets/a52f7bad-deef-4519-9737-4e1445fb9d9c)
![Image](https://github.com/user-attachments/assets/ca53fca8-3c0d-4857-a111-1e532866c6c3)
报错：
![Image](https://github.com/user-attachments/assets/140508bb-6218-4a4b-941c-5257a007e27c)

麻烦抽空帮我看看怎么解决，十分感激！
